src/App.js

```diff
@@ -1,23 +1,63 @@
-import logo from './logo.svg';
-import './App.css';
+import "./App.css";
+import React, { useState, useEffect } from "react";
+import Icon from "./components/Icon";
+
+const getQueryUrl = (city) =>
+  `${process.env.REACT_APP_API_URL}?q=${city}&units=metric&appid=${process.env.REACT_APP_API_KEY}`;
 
 function App() {
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState("");
+  const [weatherData, setWeatherData] = useState();
+  const [cityInput, setCityInput] = useState("Warszawa");
+  const [city, setCity] = useState(cityInput);
+
+  useEffect(() => {
+    setLoading(true);
+    fetch(getQueryUrl(city))
+      .then((response) => response.json())
+      .then((data) => {
+        setWeatherData(data);
+        setLoading(false);
+      })
+      .catch((e) => {
+        setLoading(false);
+        console.log("e", e);
+        setError("fetch failed");
+      });
+  }, [city]);
+
+  if (loading) {
+    return <p>loading..</p>;
+  }
+
+  if (error) {
+    return <p>ERROR: {error}</p>;
+  }
+
+  const handleKeypress = (e) => {
+    if (e.code === "Enter") {
+      setCity(cityInput);
+    }
+  };
+
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+    <div>
+      <input
+        value={cityInput}
+        onChange={(e) => setCityInput(e.target.value)}
+        onKeyPress={(e) => handleKeypress(e)}
+      />
+      <button onClick={() => setCity(cityInput)}>Submit</button>
+      <p>
+        {weatherData?.name}, {weatherData?.sys?.country}
```

Comment:
@pasternakewa create UI for two cases:
1. weatherData - is present
2. weatherData - is NOT present